### PR TITLE
CLI: Replace inquirer with prompts

### PR DESCRIPTION
**Pull request by thebuilder**
Issue: #13226 

## What I did

Replaced all instances of https://github.com/SBoudrias/Inquirer.js with https://github.com/terkelg/prompts

This is inspired by https://github.com/facebook/create-react-app/pull/10083 

They are mostly interchangeable, altho the "separator" option is missing from prompts. It can be emulated by using the "disabled" option, to prevent an item from being selected.

Based on the usage pattern, I opted for a some changes to how the prompts are presented.

**bootstrap** 
This still uses a separator.
example_1

**test**
Splits the choice in to two different questions.
example_2

**build**
Prompts you upfront if this is a watch mode or not, before presenting the list of packages.
Packages can also be selected using autocomplete.
example_3

## How to test

Run the commands relying on inquirere locally.

### CODE REVIEW

File: lib/cli/package.json [hunk 1]
```diff
@@ -52,10 +52,10 @@
     "fs-extra": "^9.0.0",
     "get-port": "^5.1.1",
     "globby": "^11.0.0",
-    "inquirer": "^7.0.0",
     "jscodeshift": "^0.6.3",
     "json5": "^2.1.1",
     "leven": "^3.1.0",
+    "prompts": "^2.0.0",
```
Comment (gaetanmaisse) [inline lib/cli/package.json:58]: You need to add `@types/prompts` to devDep as you're using `prompts` in a TS file in `lib/cli/src/initiate.ts`

**Comment by shilman** [comment]
Awesome @thebuilder! I'll merge this next week as part of 6.2-alpha. This week I'm doing hotfixes to 6.1.x and freely merging back and forth between `master` and `next`. Thanks for your contribution & patience!

**Comment by thebuilder** [comment]
Are the failing build an issue? 🙂 It's a bit overwhelming trying to understand what causes error, when it happens right after checkout.

**Comment by shilman** [comment]
@thebuilder our CI is completely borked at the moment. we're working with Circle CI to try to get it back in order. but you should only look at the required checks anyway.

**Comment by shilman** [comment]
@thebuilder Looks like there's a TS error that's breaking CI. Can you fix it?

```
@storybook/cli: src/initiate.ts(252,11): error TS2322: Type '{ type: "confirm"; name: "manual"; message: string; default: false; }' is not assignable to type 'PromptObject<"manual">'.
@storybook/cli:   Object literal may only specify known properties, and 'default' does not exist in type 'PromptObject<"manual">'.
@storybook/cli: src/initiate.ts(262,9): error TS2322: Type 'string[]' is not assignable to type 'Choice[]'.
```

**Comment by thebuilder** [comment]
@shilman Sorry about - Not sure how I could miss that. Editor might have been overloaded with types. 🙈
Corrected it, and synced with `next` branch.

**Comment by gaetanmaisse** [comment]
@thebuilder it looks like `const groups` defined in `scripts/build-package.js:L56` isn't used anymore, can you remove it?

**Pull request merged**